resources/idlharness.js

```diff
@@ -195,27 +195,60 @@ IdlArray.prototype.add_untested_idls = function(raw_idls)
 };
 
 //@}
-IdlArray.prototype.internal_add_idls = function(parsed_idls)
+IdlArray.prototype.internal_add_idls = function(parsed_idls, options)
 //@{
 {
     /**
      * Internal helper called by add_idls() and add_untested_idls().
+     *
      * parsed_idls is an array of objects that come from WebIDLParser.js's
      * "definitions" production.  The add_untested_idls() entry point
      * additionally sets an .untested property on each object (and its
      * .members) so that they'll be skipped by test() -- they'll only be
      * used for base interfaces of tested interfaces, return types, etc.
+     *
+     * options is a dictionary that can have an include or exclude member which
+     * are arrays. If include is given then only members, partials and interface
+     * targets listed will be added, and if exclude is given only those that
+     * aren't listed will be added. Only one of include and exclude can be used.
      */
+
+    if (options && options.include && options.exclude)
```

Comment:
I'll change this to only/except, per discssion in https://github.com/w3c/web-platform-tests/issues/7781